### PR TITLE
[Improvement-16516] Uniformly name process in backend code as workflow

### LINKED ISSUE #16516: [Improvement] Uniformly name process in backend code as workflow
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Brief change log

close #16516

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by ruanwenjun — DISMISSED** [review]
LGTM.

 It's better to create a DSIP to remove the upgrade code before 3.0, this means if some one want to upgrade from 2.x to 3.x, he need to download the 3.0.0-release， upgrade to 3.0 firstly.

Since the upgrade code before 3.x rely on DO object, this make us cannot change the DO.

File: dolphinscheduler-api/src/main/resources/i18n/messages_en_US.properties [hunk 1]
```diff
@@ -59,43 +52,28 @@ ALERT_PLUGIN_INSTANCE_PARAMS=alert plugin instance parameters
...
 WORKER_GROUP_NAME=worker group name
-WORKER_IP_LIST=worker ip list, eg. 192.168.1.1,192.168.1.2
 QUERY_WORKER_GROUP_PAGING_NOTES=query worker group paging
 QUERY_WORKER_GROUP_LIST_NOTES=query worker group list
 DELETE_WORKER_GROUP_BY_ID_NOTES=delete worker group by id
 DATA_ANALYSIS_TAG=analysis related operation of task state
 COUNT_TASK_STATE_NOTES=count task state
-COUNT_PROCESS_INSTANCE_NOTES=count process instance state
-COUNT_PROCESS_DEFINITION_BY_USER_NOTES=count process definition by user
+COUNT_WORKFLOW_INSTANCE_NOTES=count process instance state
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/main/resources/i18n/messages_en_US.properties:65]: Good job, you do not missing those i18n files

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 2]
```diff
@@ -16,22 +16,21 @@
...
-    `dept_project_code`            bigint   NOT NULL DEFAULT 0 COMMENT 'dependent project code',
-    `dept_process_definition_code` bigint   NOT NULL DEFAULT 0 COMMENT 'dependent process definition code',
-    `dept_task_definition_code`    bigint   NOT NULL DEFAULT 0 COMMENT 'dependent task definition code',
-    `create_time`                  datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'create time',
-    `update_time`                  datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'update time',
-    PRIMARY KEY (`id`),
-    KEY                            `idx_process_code_version` (`process_definition_code`,`process_definition_version`),
-    KEY                            `idx_task_code_version` (`task_definition_code`,`task_definition_version`),
-    KEY                            `idx_dept_code` (`dept_project_code`,`dept_process_definition_code`,`dept_task_definition_code`)
+CREATE TABLE `t_ds_process_task_lineage` (
```
Comment (zhongjiajie) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql:19]: So our strategy for database schema is to change only the unreleased version but not the newly released version?

**Review by zhongjiajie — DISMISSED** [review]
LGTM overall, thanks for your contribute, it takes some times to change all of this

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql (hunk 2 above)
Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql:19]: > So our strategy for database schema is to change only the unreleased version but not the newly released version?

I'll do that in the next step in #16525 .

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by zhongjiajie — APPROVED** [review]
LGTM

**Comment by EricGao888** [comment]
Huge improvement! Glad to see this PR merged. Thx @SbloodyS !

**Pull request merged**